uprotocol_vsomeip/vsomeip_utransport.py

```diff
@@ -0,0 +1,494 @@
+"""
+SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
+
+See the NOTICE file(s) distributed with this work for additional
+information regarding copyright ownership.
+
+This program and the accompanying materials are made available under the
+terms of the Apache License Version 2.0 which is available at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+SPDX-License-Identifier: Apache-2.0
+"""
+import json
+import logging
+import os
+import socket
+import sys
+import threading
+import time
+from builtins import str
+from concurrent.futures import Future
+from enum import Enum
+from typing import Tuple, Final
+
+from vsomeip_py import vsomeip
+from uprotocol.proto.uattributes_pb2 import CallOptions, UMessageType, UPriority
+from uprotocol.proto.umessage_pb2 import UMessage
+from uprotocol.proto.upayload_pb2 import UPayload, UPayloadFormat
+from uprotocol.proto.uri_pb2 import UEntity, UUri
+from uprotocol.proto.ustatus_pb2 import UStatus, UCode
+from uprotocol.rpc.rpcclient import RpcClient
+from uprotocol.transport.builder.uattributesbuilder import UAttributesBuilder
+from uprotocol.transport.ulistener import UListener
+from uprotocol.transport.utransport import UTransport
+from uprotocol.uri.validator.urivalidator import UriValidator
+from uprotocol.uuid.serializer.longuuidserializer import LongUuidSerializer
+
+from .helper import VsomeipHelper
+
+_logger = logging.getLogger("vsomeip_utransport" + "." + __name__)
+IS_WINDOWS: Final = sys.platform.startswith("win")
+
+
+class VsomeipTransport(UTransport, RpcClient):
+    """
+    Vsomeip Transport
+    """
+
+    _futures = {}
+    _registers = {}
+    _responses = {}
+    _instances = {}
+    _configuration = {}
+    _requests = {}
+    _published = {}
+    _lock = threading.Lock()
+
+    INSTANCE_ID: Final = 0x0000
+    MINOR_VERSION: Final = 0x0000
+
+    class VSOMEIPType(Enum):
+        """
+        Types of VSOMEIP Application Server/Client
+        """
+
+        CLIENT = "Client"
+        SERVICE = "Service"
+
+    def __init__(
+        self,
+        source: UUri = UUri(),
```

Comment:
Wasn't sure initially if it would come from uri.authority.ip, hence did not provide the option earlier to edit unicast. 

Changed now to be able to input unicast as well